### PR TITLE
[ARCTIC-1204][AMS] Start a new scheduled thread to refresh tables

### LINKED ISSUE #1204: [Improvement]: Optimize Check Worker is busy performing operations to update the state
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Optimize Check Worker is busy performing operations to update the optimize state, resulting in a delay in refreshing the table list

### How should we improve?

Split optimize check task and refresh tables task. In other word, start a new scheduled thread in `OptimizeService` that refresh tables in all catalogs in fixed interval.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?
fix #1204 

## Brief change log
When a brand-new table/enable self-optimizing table is added or a exsting table is dropped, AMS can update the managed table list in a timely manner.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java [hunk 1]
```diff
@@ -76,6 +82,8 @@ public class OptimizeService extends IJDBCService implements IOptimizeService {
 
   private ScheduledTasks<TableIdentifier, OptimizeCheckTask> checkTasks;
 
+  private ScheduledExecutorService refreshTableEsv;
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java:85]: May be we don't need this field

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java (hunk 1 above)
Comment (XBaith) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java:85]: Make sense

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
Thanks for your contributions

**Comment by wangtaohz** [comment]
LGTM

**Pull request merged**